test/integration/features/addons/addon-htpasswd-identity-provider.feature

```diff
@@ -0,0 +1,86 @@
+@addon-htpasswd-identity-provider @core
+Feature: Configures single identity for OpenShift login instance
+This add-on configures OpenShift to use only a single identity provider
+
+  Scenario: User can enable the htpasswd-identity-provider
+     When executing "minishift addons enable htpasswd-identity-provider" succeeds
+     Then stdout should contain "Add-on 'htpasswd-identity-provider' enabled"
+
+  Scenario: User can start Minishift with htpasswd-identity-provider add-on enabled
+    Given Minishift has state "Does Not Exist"
+      And image caching is disabled
+     When executing "minishift start" succeeds
+     Then Minishift should have state "Running"
+      And stdout should contain
+      """
+      Successfully installed addon htpasswd identity provider
+      """
+
+  Scenario: User can login to openshift instance only with default username "developer" and password "developer"
+    Given Minishift has state "Running"
+      And user waits "60" seconds
+     When executing "oc login -u developer -p developer" succeeds
+     Then stdout should contain
+      """
+      Login successful.
+      """
+     When executing "oc login -u developer -p anything" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+     When executing "oc login -u anything -p developer" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+    
+  Scenario: User can update login password for default user "developer"
+    Given Minishift has state "Running"
+     When executing "minishift addons apply htpasswd-identity-provider --addon-env USERNAME=developer --addon-env USER_PASSWORD=password" succeeds
+      And user waits "60" seconds
+     Then executing "oc login -u developer -p password" succeeds
+      And stdout should contain
+      """
+      Login successful.
+      """
+     When executing "oc login -u developer -p developer" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+     When executing "oc login -u developer -p anything" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+  
+  Scenario: User can create login credentials for openshift instance and can login using that credentials
+    Given Minishift has state "Running"
+     When executing "minishift addons apply htpasswd-identity-provider --addon-env USERNAME=openshift-user --addon-env USER_PASSWORD=openshift-password" succeeds
```

Comment:
does `developer:developer` still exist and is usable?

Reply:
Yes, AFAIU the default user still present is the cluster, so it can be logged to the default namespace ```myproject``` with any password if not set through ```--addon-env```

Reply:
> does developer:developer still exist and is usable?

@gbraad yes it does exist, what this step does is just add another user to htpasswd file which is used to authenticate to console.

Reply:
This is what I expect, so i would like to see this enforced in the test (as this acts like a precaution that NO clear action is performed in future) and to 'document' this behaviour.

Reply:
@gbraad you mean we need to verify the file htpasswd to cross check.

Reply:
@gbraad I have added a scenario ```Scenario: User can still use default user developer's updated credentials to login openshift instance``` just after the ```Scenario: User can login using new credentials```. This will also enfource behavior you are looking for. Please share if you have any other thought.

Reply:
Right. We needed to check if the exsiting users are still able to be used.

Reply:
Accidentally my comment was deleted just above gbraad comment ```Right. We needed to check if the exsiting users are still able to be used.```

Adding the same comment again

I have added a scenario Scenario: User can still use default user developer's updated credentials to login openshift instance just after the Scenario: User can login using new credentials. This will also enfource behavior you are looking for. Please share if you have any other thought.